nemo/collections/common/metrics/per.py

```diff
@@ -0,0 +1,428 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from collections import namedtuple
+from tqdm import tqdm
+
+from nemo.utils import logging
+
+
+def punctuation_error_rate(
+    references: list[str], hypotheses: list[str], punctuation_marks: list[str], punctuation_mask: str = "[PUNCT]",
+) -> None:
+
+    """
+    Computes Punctuation Error Rate
+    
+    Args:
+        references (list[str]) - list of references
+        hypotheses (list[str]) - list of hypotheses
+        punctuation_marks (list[str]) - list of punctuation marks for computing metrics
+        punctuation_mask (str, by default "[PUNCT]") - mask token that will be applied to
+        given punctuation marks while edit distance calculation
+        
+    Return:
+        per (float) - Punctuation Error Rate
+    """
+
+    per_data_obj = PERData(
+        references=references,
+        hypotheses=hypotheses,
+        punctuation_marks=punctuation_marks,
+        punctuation_mask=punctuation_mask,
+    )
+
+    per_data_obj.compute()
+
+    return per_data_obj.per
+
+
+class PER:
```

Comment:
Maybe name this to PuncER or something like that to denote it's different from phoneme error rate

Reply:
Thank you, I have fixed it with commit [c01baad](https://github.com/NVIDIA/NeMo/pull/7538/commits/c01baad459747978e1847cf4a2cdd6c21ee637d5)